src/flamegraph/mod.rs

```diff
@@ -608,33 +649,21 @@ where
     Ok(())
 }
 
-/// Produce a flame graph from a reader that contains a sorted sequence of folded stack lines.
+/// Produce a flame graph from a reader that contains a sequence of folded stack lines.
 ///
 /// See [`from_sorted_lines`] for the expected format of each line.
 ///
 /// The resulting flame graph will be written out to `writer` in SVG format.
-pub fn from_reader<R, W>(opt: &mut Options, mut reader: R, writer: W) -> quick_xml::Result<()>
+pub fn from_reader<R, W>(opt: &mut Options, reader: R, writer: W) -> quick_xml::Result<()>
 where
     R: Read,
     W: Write,
 {
-    let mut input = String::new();
-    reader
-        .read_to_string(&mut input)
-        .map_err(quick_xml::Error::Io)?;
-
-    if opt.no_sort {
-        from_sorted_lines(opt, input.lines(), writer)
-    } else {
-        let mut lines: Vec<&str> = input.lines().collect();
-        lines.sort_unstable();
-        from_sorted_lines(opt, lines, writer)
-    }
+    from_readers(opt, iter::once(reader), writer)
```

Comment:
I _think_ this change means that even with `no_sort`, the entire file gets read into memory before we do any processing. That seems a little unfortunate, though maybe it's a cost we're willing to pay? It means that if I have a _huge_ trace file (like, many GB, which isn't unreasonable with debug symbols) that I know is sorted, I may be unable to plot it at all since it won't fit in memory all at once.

Reply:
Good point, I completely overlooked that.

Reply:
It looks like it was already reading the entire file into memory before the change. I guess there are a couple of reasons for this.
- `from_reader` takes a `Read`, not a `BufRead`
- Even if it took a `BufRead`, the `lines` method returns an iterator that produces `io::Result<String>`, so it can't be passed to `from_lines`.

Reply:
I guess we require `Iterator<Item = &str>`, so that still assumes you have all the strings still lying around somewhere. This may be worth thinking harder about, but not relevant to this PR :thinking: